lib/Library.php

```diff
@@ -16,95 +16,68 @@
 class Library{
 	/**
 	* @param array<string> $tags
-	* @return array<Ebook>
-	* @throws Exceptions\AppException
+	* @return array<string, array<Ebook>|int>
 	*/
-	public static function FilterEbooks(string $query = null, array $tags = [], EbookSortType $sort = null): array{
-		$ebooks = Library::GetEbooks();
-		$matches = $ebooks;
+	public static function FilterEbooks(string $query = null, array $tags = [], EbookSortType $sort = null, int $page = 1, int $perPage = EBOOKS_PER_PAGE): array{
+		// Returns an array of:
+		// ['ebooks'] => array<Ebook>,
+		// ['ebooksCount'] => int
```

Comment:
You can document this officially in the function's PHPDoc with Array Shapes: https://phpstan.org/writing-php-code/phpdoc-types#array-shapes

Reply:
Much better, thanks for the guidance! I'll do the same to `FilterArtwork()` in a follow-up change.